### DISCUSSION TITLE
Filament Tabs

### DISCUSSION THREAD

**Discussion by ShakerAwad74** [filamentphp/filament#13026] [Help]
### Package

Panel builder

### Package Version

v3

### How can we help you?

How can I create a vertical tab in Filament? I want to create a vertical tab like the one in the image below.
![image](https://github.com/filamentphp/filament/assets/121998317/605103f9-8cb4-47d8-a032-8bd800da0cdf)

**Comment by dissto** [comment]
Probably a [cluster](https://filamentphp.com/docs/3.x/panels/clusters#overview) or a [resource sub navigation](https://filamentphp.com/docs/3.x/panels/resources/getting-started#resource-sub-navigation) are best suited for that as a starting point? 🤔

**Comment by Bender-0** [comment]
i needed something similar , this worked for me:

```html
<div class="flex h-screen" x-data="{ activeTab: 'tab1' }">
    <div class="flex flex-col w-64">
        <x-filament::tabs class="flex flex-col justify-start w-full space-y-4">
            <x-filament::tabs.item alpine-active="activeTab === 'tab1'" x-on:click="activeTab = 'tab1'">
                Tab 1
            </x-filament::tabs.item>

            <x-filament::tabs.item alpine-active="activeTab === 'tab2'" x-on:click="activeTab = 'tab2'">
                Tab 2
            </x-filament::tabs.item>
        </x-filament::tabs>
    </div>

    <div class="flex-1 p-4">
        <div x-show="activeTab === 'tab1'">
            Content of tab1
        </div>

        <div x-show="activeTab === 'tab2'">
            Content of tab2
        </div>
    </div>
</div>

**Reply by mikemastercorp** [reply]
Hey @Bender-0 Bender. where did you place your markup to achieve the vertical tabs as such? have you replaced the original filament tabs code or you've created a custom view and used that as a Livewire component inside?

**Reply by Bender-0** [reply]
Hi @mikemastercorp i needed vertical tabs in the profile page so i created custom livewire component and used the original filament tabs blade component with tailwind to align them vertically.

**Reply by mikemastercorp** [reply]
I see @Bender-0 

I am tempted to create a Filament Starter Pack like the Super Duper one that exists which I could not test as it exploded in loading timeout and I can't get onhold, so my idea is to have multiple vertical tabs that allow some Settings to be saved/stored in the database using Spatie settings. For sure there will be more to come as modules and goodies, but I thought of using tabs as an easy implementation, but other things might be to use Navigation inside the page too (if I manage to get it to work). Long story short, the vertical tabs are needed to allow better organization of the Settings and eventually allow a single settings page to store all application configurations. And yes, I can use a navigation group with multiple settings 9as it is now) but It is strange to be clicking from page to page to change some parameters, and I am learning Filament still so trying to get tough tasks resolved and learn on the way ;) ).

Your option should work too, so I can try with a custom Livewire module + new page to achieve that if we have no option for now to support vertical tabs...

**Reply by Bender-0** [reply]
Hey @danharrin is this suitable for the framework? your input on this would be greatly appreciated.

@mikemastercorp i have implemented vertical method on Tabs component:
this is the default tabs component
![Screenshot 2024-09-30 225048](https://github.com/user-attachments/assets/9441d8df-d975-4b3e-995b-8791affeb21c)
![Screenshot 2024-09-30 224954](https://github.com/user-attachments/assets/66f8f4ac-0e64-4212-9eb7-5b467fe0fa36)

and this is with vertical() method implementation and contained(true):
![Screenshot 2024-09-30 225422](https://github.com/user-attachments/assets/c817f704-5764-48c5-914a-a7a1ce627cf9)
![Screenshot 2024-09-30 225207](https://github.com/user-attachments/assets/42ee54e4-c97e-4ab3-9df0-d8aaa55cc91b)
![Screenshot 2024-09-30 225224](https://github.com/user-attachments/assets/abf68210-e08a-47cb-b8bb-adc4c00a7f93)
 
and this is with vertical() and contained(false)
![Screenshot 2024-09-30 225441](https://github.com/user-attachments/assets/1c0008b2-926e-462b-a090-325e6551fdbc)
![Screenshot 2024-09-30 225504](https://github.com/user-attachments/assets/2ce4140b-b389-4893-8b71-fbe24964669e)
![Screenshot 2024-09-30 225518](https://github.com/user-attachments/assets/81a4ca80-a6a6-4211-9a18-b298e909d846)

i want to submit a PR but i feel the UI needs some improvement.. any suggestions?

**Reply by danharrin** [reply]
I like the concept, so happy to see a PR and then we can handle it from there! Thanks!

Please make sure it is based on v4 and not v3, if you want to wait for v4 to be released that is also fine.

**Reply by Bender-0** [reply]
@mikemastercorp if you like to implement it in your project i can provide the code
https://github.com/filamentphp/filament/pull/14436

**Comment by mikemastercorp** [comment]
I second the need of vertical tabs and I was left with the impression that there was a method called: vertical() to allow vertical tabs alignment but it seems like my filament forms (latest 3.x version with Laravel 11) did not recognize it as a valid method.

I know we can edit the CSS classes to re-align tabs vertically etc. however it seems like a great improvement to me to allow two parameters:
1. to accept the tabs alignment - horizontal/vertical
2. To allow the dev to define the tabs location (left, right, center, top, bottom). If not left/right as we can have specific classes for that with Tailwind, at least to have top/bottom placement of tabs as sometimes a design/UI idea might be to have the tabs vertically bottom aligned or horizontally vs default top centered location.

**Comment by rasul-chopurov** [comment]
https://github.com/AFS19/filament-vertical-tabs
